go/vt/tabletserver/sqlquery.go

```diff
@@ -479,27 +530,32 @@ func (sq *SqlQuery) SplitQuery(ctx context.Context, req *proto.SplitQueryRequest
 	return nil
 }
 
-// startRequest validates the current state and sessionId and registers
+// startRequest validates the current state and sessionID and registers
 // the request (a waitgroup) as started. Every startRequest requires one
 // and only one corresponding endRequest. When the service shuts down,
 // disallowQueries will wait on this waitgroup to ensure that there are
 // no requests in flight.
-func (sq *SqlQuery) startRequest(sessionId int64, allowShutdown bool) (err error) {
-	sq.mu.RLock()
-	defer sq.mu.RUnlock()
-	st := sq.state.Get()
-	if st == SERVING {
+// ignoreSession is passed in as true for valid internal requests that don't have a session id.
+func (sq *SqlQuery) startRequest(sessionID int64, ignoreSession, allowShutdown bool) (err error) {
+	sq.mu.Lock()
```

Comment:
From our offline discussion, I now understand why this is an exclusive lock as opposed to a Read lock. Please add a comment for the reasoning here.